packages/@react-spectrum/dropzone/docs/DropZone.mdx

```diff
@@ -0,0 +1,374 @@
+
+function Example() {
+  let [isFilled, setIsFilled] = React.useState(false);
+  let [filledSrc, setFilledSrc] = React.useState(null);
+
+  return (
+    <>
+      <Draggable />
+      <DropZone
+        maxWidth="size-3000"
+        isFilled={isFilled}
+        onDrop={async (e) => {
+          e.items.find(async (item) => {
+            if (item.kind === 'file') {
+              setFilledSrc(item.name);
+              setIsFilled(true);
+              
+            } else if (item.kind === 'text' && item.types.has('text/plain')) {
+              setFilledSrc(await item.getText('text/plain'));
+              setIsFilled(true);
+            }
+          });
+        }}>
+        <IllustratedMessage>
+          <Upload />
+          <Heading>
+            <Text slot="label">
+              Drag and drop here
+            </Text>
+          </Heading>
+        </IllustratedMessage>
+      </DropZone>
+      {isFilled && 
+        <div className="files">
+          <File />
+          {filledSrc}
+        </div>}
+    </>
+  );
+}
+```
+
+The `Draggable` component used above is defined below. See [useDrag](../react-aria/useDrag.html) for more details and documentation.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example render=false export=true
+import {useDrag} from '@react-aria/dnd';
+
+function Draggable() {
+  let {dragProps, isDragging} = useDrag({
+    getItems() {
+      return [{
+        'text/plain': 'hello world',
+        'my-app-custom-type': JSON.stringify({message: 'hello world'})
+      }];
+    }
+  });
+
+  return (
+    <div {...dragProps} role="button" tabIndex={0} className={`draggable ${isDragging ? 'dragging' : ''}`}>
+      Drag me
+    </div>
+  );
+}
+```
+</details>
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.draggable {
+  display: inline-block;
+  vertical-align: top;
+  border: 1px solid gray;
+  padding: 10px;
+  margin-right: 20px;
+  margin-bottom: 20px;
+  border-radius: 4px;
+  height: fit-content;
+}
+
+.draggable.dragging {
+  opacity: 0.5;
+}
+
+.files{
+  margin-top: 20px;
+}
+```
+
+</details>
+
+
+## Props
+
+<PropTable component={docs.exports.DropZone} links={docs.links} />
+
+## Visual options
+
+### Filled state
+
+The user is responsible for both managing the filled state of a drop zone and handling the associated styling. To set the drop zone to a filled state, the user must pass the `isFilled` prop.
+
+The example below demonstrates one way of styling the filled state.
+
+```tsx example
+function Example() {
+  let [filledSrc, setFilledSrc] = React.useState(null);
+  let [isFilled, setIsFilled] = React.useState(false);
+
+  return (
+    <>
+      <DraggableImage />
+      <DropZone
+        isFilled={isFilled}
+        maxWidth="size-3000"
+        height="size-2400"
+        UNSAFE_className={isFilled && 'is-filled'}
```

Comment:
I don't think we want to encourage `UNSAFE_className` anywhere in the docs. Is this style needed? And if so, could we include it automatically somehow?

Reply:
it's not required by any means, mostly just aesthetic. i thought that it looked nicer when the border was solid versus dashed when the image filled up the dropzone area. i can try some other ways and see it goes but in the worst-case, i don't think removing this line will have any large effect

Reply:
good catch, bring up in standup so we can determine if we want to build this into the component or if we need to bring it back to design
whatever the outcome, we should not use UNSAFE_className here